Tests/UnitTests/Core/Vertexing/LinearizedTrackFactoryTests.cpp

```diff
@@ -58,137 +67,60 @@ std::uniform_real_distribution<> resAngDist(0., 0.1);
-  double z0v = z;
-
-  // Start constructing nTracks tracks in the following
-  std::vector<BoundTrackParameters> tracks;
-
-  // Construct random track emerging from vicinity of vertex position
-  // Vector to store track objects used for vertex fit
-  for (unsigned int iTrack = 0; iTrack < nTracks; iTrack++) {
-    // Construct positive or negative charge randomly
-    double q = qDist(gen) < 0 ? -1. : 1.;
-
-    // Construct random track parameters
-    BoundVector paramVec;
-    paramVec << d0v + d0Dist(gen), z0v + z0Dist(gen), phiDist(gen),
-        thetaDist(gen), q / pTDist(gen), 0.;
-
-    // Resolutions
-    double resD0 = resIPDist(gen);
-    double resZ0 = resIPDist(gen);
-    double resPh = resAngDist(gen);
-    double resTh = resAngDist(gen);
-    double resQp = resQoPDist(gen);
-
-    // Fill vector of track objects with simple covariance matrix
-    Covariance covMat;
-
-    covMat << resD0 * resD0, 0., 0., 0., 0., 0., 0., resZ0 * resZ0, 0., 0., 0.,
-        0., 0., 0., resPh * resPh, 0., 0., 0., 0., 0., 0., resTh * resTh, 0.,
-        0., 0., 0., 0., 0., resQp * resQp, 0., 0., 0., 0., 0., 0., 1.;
-    tracks.emplace_back(perigeeSurface, paramVec, std::move(covMat));
-  }
-
-  Linearizer::Config ltConfig(bField, propagator);
-  Linearizer linFactory(ltConfig);
-  Linearizer::State state(bField->makeCache(magFieldContext));
-
-  BoundVector vecBoundZero = BoundVector::Zero();
-  BoundSymMatrix matBoundZero = BoundSymMatrix::Zero();
-  Vector4 vecSPZero = Vector4::Zero();
-  ActsMatrix<eBoundSize, 4> matBound2SPZero = ActsMatrix<eBoundSize, 4>::Zero();
-  ActsMatrix<eBoundSize, 3> matBound2MomZero =
-      ActsMatrix<eBoundSize, 3>::Zero();
-
-  for (const BoundTrackParameters& parameters : tracks) {
-    LinearizedTrack linTrack =
-        linFactory
-            .linearizeTrack(parameters, Vector4::Zero(), geoContext,
-                            magFieldContext, state)
-            .value();
-
-    BOOST_CHECK_NE(linTrack.parametersAtPCA, vecBoundZero);
-    BOOST_CHECK_NE(linTrack.covarianceAtPCA, matBoundZero);
-    BOOST_CHECK_EQUAL(linTrack.linearizationPoint, vecSPZero);
-    BOOST_CHECK_NE(linTrack.positionJacobian, matBound2SPZero);
-    BOOST_CHECK_NE(linTrack.momentumJacobian, matBound2MomZero);
-    BOOST_CHECK_NE(linTrack.constantTerm, vecBoundZero);
+  // Set up stepper and propagator for 0 B-field
+  StraightLineStepper straightStepper;
+  auto straightPropagator =
+      std::make_shared<Propagator<StraightLineStepper>>(straightStepper);
+
+  // Create perigee surface, initial track parameters will be relative to it
+  std::shared_ptr<PerigeeSurface> perigeeSurface{
+      Surface::makeShared<PerigeeSurface>(Vector3{0., 0., 0.})};
+
+  // Vertex position and corresponding d0 and z0
+  Vector4 vtxPos;
+  double d0v;
+  double z0v;
+  {
+    double x = vXYDist(gen);
+    double y = vXYDist(gen);
+    double z = vZDist(gen);
+    vtxPos << x, y, z, 0;  // TODO: add time
+    d0v = std::hypot(x, y);
+    z0v = z;
```

Comment:
this was the case before but I think it is odd to use random values for a single case

Reply:
I agree, but I guess one could play around with the seed to see how the linearization performs for different (random) vertices.. Should I remove it and hardcode something here nonetheless?

Reply:
lets leave it as is is